Python/Product/Cookiecutter/View/CookiecutterOptionsPage.xaml.cs

```diff
@@ -76,7 +77,28 @@ class TemplateContextItemTemplateSelector : DataTemplateSelector {
             var element = container as FrameworkElement;
             var p = item as ContextItemViewModel;
             if (element != null && p != null) {
-                var templateName = p.Items.Count == 0 ? "textTemplate" : "listTemplate";
+                string templateName;
+                switch (p.ValueType) {
+                    case Model.ContextItemValueType.String:
+                        templateName = "textTemplate";
```

Comment:
Cap the length of the selector name (30 chars seems like enough), make it lowercase and then use `Regex.IsMatch(selector, "^[a-z]+$")` to check that it's safe first.